### PR TITLE
Ewb 4089 build docs as part of all workflows

### PR DESCRIPTION
**Pull request by alourie**
# Description

Doc building improvements

### CODE REVIEW

File: .github/workflows/maven-build.yml [hunk 1]
```diff
@@ -72,3 +72,36 @@ jobs:
...
+        run: | 
+          rm -rf ~/.npmrc
+          echo "@zepben:registry=${{ secrets.NEXUS_NPM_REPO }}" >> ~/.npmrc
+          echo "//mavenrepo.zepben.com/repository/zepben-npm/:_authToken=${{ secrets.CI_NPM_TOKEN }}" >> ~/.npmrc
+          echo "//npm.pkg.github.com/:_authToken=${{ secrets.CI_GITHUB_TOKEN }}" >> ~/.npmrc
+          echo "\n" >> ~/.npmrc
+
+      - name: Build docs
+        run: |
+          if [ -d docs ]; then
```
Comment (kgreav) [inline .github/workflows/maven-build.yml:103]: this check could be done as part of build-and-test, but you could just save the result as an output, and then you should be able to apply the if condition to the build-docs job. That will save us running the container and a bunch of steps just to find out we don't have any docs. Can you make this change in each build?

**Review by kgreav — CHANGES_REQUESTED** [review]

File: .github/workflows/maven-build.yml (hunk 1 above)
Comment (alourie) [reply .github/workflows/maven-build.yml:103]: yup, can do

File: .github/workflows/csharp-build.yml [hunk 2]
```diff
@@ -46,6 +46,8 @@ on:
 jobs:
   build:
```
Comment (kgreav) [inline .github/workflows/csharp-build.yml:47]: can you make this one build-and-test for consistency plz

**Review by kgreav — CHANGES_REQUESTED** [review]

File: .github/workflows/csharp-build.yml (hunk 2 above)
Comment (alourie) [reply .github/workflows/csharp-build.yml:47]: :+1:

File: .github/workflows/maven-build-oss.yml [hunk 3]
```diff
@@ -36,3 +38,31 @@ jobs:
...
+      - uses: actions/checkout@v3
+
+      - name: Build docusaurus
+        id: build
+        uses: zepben/docusaurus-action@main
+        with:
+          TAG-AND-RELEASE: false
+          NPM_REPO: ${{ secrets.NPM_REPO }}
+          NPM_TOKEN: ${{ secrets.CI_GITHUB_TOKEN }}
+        continue-on-error: true
```
[the remark this answers is not in the document]
Comment (alourie) [reply .github/workflows/maven-build-oss.yml:68]: yea will do

**Review by kgreav — APPROVED** [review]

**Pull request merged**